library/java/io/envoyproxy/envoymobile/engine/JvmFilterContext.java

```diff
@@ -47,79 +47,85 @@ public void passTrailer(byte[] key, byte[] value, boolean start) {
-   * @param ignored,     StreamIntel not yet supported by filter callbacks.
+   * @param streamIntel,     StreamIntel not yet supported by filter callbacks.
    * @return Object[],   pair of HTTP filter status and optional modified headers.
    */
-  public Object onRequestHeaders(long headerCount, boolean endStream, long[] ignored) {
+  public Object onRequestHeaders(long headerCount, boolean endStream, long[] streamIntel) {
     assert headerUtility.validateCount(headerCount);
     final Map headers = headerUtility.retrieveHeaders();
-    return toJniFilterHeadersStatus(filter.onRequestHeaders(headers, endStream));
+    return toJniFilterHeadersStatus(
+        filter.onRequestHeaders(headers, endStream, new EnvoyStreamIntelImpl(streamIntel)));
   }
 
   /**
    * Dispatches data received from the JNI layer up to the platform.
    *
    * @param data,      chunk of body data from the HTTP request.
    * @param endStream, indicates this is the last remote frame of the stream.
-   * @param ignored,   StreamIntel not yet supported by filter callbacks.
+   * @param streamIntel,   StreamIntel not yet supported by filter callbacks.
    * @return Object[], pair of HTTP filter status and optional modified data.
    */
-  public Object onRequestData(byte[] data, boolean endStream, long[] ignored) {
+  public Object onRequestData(byte[] data, boolean endStream, long[] streamIntel) {
     ByteBuffer dataBuffer = ByteBuffer.wrap(data);
-    return toJniFilterDataStatus(filter.onRequestData(dataBuffer, endStream));
+    return toJniFilterDataStatus(
+        filter.onRequestData(dataBuffer, endStream, new EnvoyStreamIntelImpl(streamIntel)));
   }
 
   /**
    * Invokes onTrailers callback using trailers passed via passHeaders.
    *
    * @param trailerCount, the total number of trailers included in this header block.
-   * @param ignored,      StreamIntel not yet supported by filter callbacks.
+   * @param streamIntel,      StreamIntel not yet supported by filter callbacks.
    * @return Object[],    pair of HTTP filter status and optional modified trailers.
    */
-  public Object onRequestTrailers(long trailerCount, long[] ignored) {
+  public Object onRequestTrailers(long trailerCount, long[] streamIntel) {
     assert headerUtility.validateCount(trailerCount);
     final Map trailers = headerUtility.retrieveHeaders();
-    return toJniFilterTrailersStatus(filter.onRequestTrailers(trailers));
+    return toJniFilterTrailersStatus(
+        filter.onRequestTrailers(trailers, new EnvoyStreamIntelImpl(streamIntel)));
   }
 
   /**
    * Invokes onHeaders callback using headers passed via passHeaders.
    *
    * @param headerCount, the total number of headers included in this header block.
    * @param endStream,   whether this header block is the final remote frame.
-   * @param ignored,     StreamIntel not yet supported by filter callbacks.
+   * @param streamIntel,     StreamIntel not yet supported by filter callbacks.
```

Comment:
```suggestion
   * @param streamIntel, StreamIntel not yet supported by filter callbacks.
```
not sure whether this comment is still valid